internal/pkg/cosign/fulcio/attestor.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2021 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fulcio
+
+import (
+	"context"
+	"crypto"
+	"io"
+
+	"github.com/sigstore/cosign/internal/pkg/cosign"
+	"github.com/sigstore/cosign/pkg/oci"
+	"github.com/sigstore/cosign/pkg/oci/mutate"
+)
+
+// fulcioAttestor still needs to actually upload keys to Fulcio and receive
+// the resulting `Cert` and `Chain`, which are added to the returned `oci.Signature`
+type fulcioAttestor struct {
+	inner cosign.DSSEAttestor
+
+	cert, chain []byte
+}
+
+var _ cosign.DSSEAttestor = (*fulcioAttestor)(nil)
+
+// Attest implements `cosign.DSSEAttestor`
+func (fa *fulcioAttestor) DSSEAttest(ctx context.Context, payload io.Reader) (oci.Signature, crypto.PublicKey, error) {
+	att, pub, err := fa.inner.DSSEAttest(ctx, payload)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// TODO(dekkagaijin): move the fulcio SignerVerififer logic here
```

Comment:
Can we just do this?

I feel like we keep rehashing the argument about `fulcio` being a wrapper 😉

Reply:
I would, but doing that involves changing _everything_

Reply:
Only if you *use* it right?  We talked about forking chunks of logic over here and then replacing the spaghetti monster with the composable blocks 😇

Reply:
fair, I'll migrate things over for the Attestor